src/resolveRequestDocument.ts

```diff
@@ -36,7 +36,19 @@ export const resolveRequestDocument = (
     return { query: document, operationName }
   }
 
-  const operationName = extractOperationName(document)
-
-  return { query: print(document), operationName }
+  const fragmentsNames: Set<string> = new Set()
+  const fragmentsDedupedDocument = visit(document, {
+    FragmentDefinition: (node) => {
+      const fragmentName = `${node.name.value}-${node.typeCondition.name.value}`
+      if (fragmentsNames.has(fragmentName)) {
+        return null
+      }
+      fragmentsNames.add(fragmentName)
+      return undefined
+    },
+  })
+
+  const operationName = extractOperationName(fragmentsDedupedDocument)
+
+  return { query: print(fragmentsDedupedDocument), operationName }
```

Comment:
Want to understand this better.

Which part of the system is responsible for duplicating the fragments in the first place and what problems does it cause?

Reply:
The problem is solved by this pull request dotansimha/graphql-code-generator#8971 and I think this pull request is no longer needed, unless someone use `...` spread for fragment definitions within a document node then I don't know if we really need to handle this case here or the user should.

Reply:
`graphql-request`'s job is not to clean up user's bad input.